clang-tools-extra/clangd/index/SymbolCollector.cpp

```diff
@@ -409,7 +409,7 @@ class SymbolCollector::HeaderFileURICache {
     // Framework headers are spelled as <FrameworkName/Foo.h>, not
     // "path/FrameworkName.framework/Headers/Foo.h".
     auto &HS = PP->getHeaderSearchInfo();
-    if (const auto *HFI = HS.getExistingFileInfo(*FE, /*WantExternal*/ false))
+    if (const auto *HFI = HS.getExistingFileInfo(*FE))
```

Comment:
Shouldn't this be `getExistingLocalFileInfo` since it was previously `/*WantExternal*/ false`?

Reply:
I think that would be closer to the original semantics, but my feeling is that it's not correct. This `FileEntry` corresponds to a `FileID` containing the `SourceLocation` of a `NamedDecl` which (I think) might've been deserialized from a PCM file. So I think including external `HeaderFileInfo` here is the right thing to do.

Reply:
Could this code be performance sensitive?

Reply:
Might be, I'm not sure. CC @DavidGoldman

Reply:
Changed this to `getExistingLocalFileInfo()` and will create a separate PR for changing this to `getExistingFileInfo()`.

Reply:
https://github.com/llvm/llvm-project/pull/88446